packages/mermaid/src/rendering-util/splitText.ts

```diff
@@ -0,0 +1,128 @@
+import type { CheckFitFunction, MarkdownLine, MarkdownWord, MarkdownWordType } from './types.js';
+
+/**
+ * Splits a string into graphemes if available, otherwise characters.
+ */
+export function splitTextToChars(text: string): string[] {
+  if (Intl.Segmenter) {
```

Comment:
Hmmmmm, can we somehow mock that `Intl.Segmenter` doesn't exist in Vitest and/or Cypress?

I guess with Cypress, we could enable a Firefox run somehow, although that would be very slow :)

Reply:
I'm not sure about testing it in multiple browsers, but sure there's some differences, see https://github.com/mermaid-js/mermaid/pull/4514#discussion_r1235560767.

Reply:
I've added unit tests without `Intl.Segmenter`. As the checkFit function is abstracted out, I guess we can get by without testing in a different browser, for now. 

But it would be good to have tests in Chrome and FF in the future.

Reply:
> I've added unit tests without `Intl.Segmenter`. As the checkFit function is abstracted out, I guess we can get by without testing in a different browser, for now. 

Sounds good to me!

It was thanks to the new Codecov action that I even noticed this issue :)